bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/dto/ConsumptionReport.java

```diff
@@ -34,6 +41,32 @@
         public List<String> labels;
         public List<Period> periodes;
         public List<Double> datas;
+
+        public void log(String title, boolean withDateFin, DateTimeFormatter dateTimeFormatter, boolean onlyLast) {
+            if (LOGGER.isDebugEnabled()) {
+                int size = (datas == null || periodes == null) ? 0
+                        : (datas.size() <= periodes.size() ? datas.size() : periodes.size());
+                if (onlyLast) {
+                    if (size > 0) {
+                        log(size - 1, title, withDateFin, dateTimeFormatter);
+                    }
+                } else {
+                    for (int i = 0; i < size; i++) {
+                        log(i, title, withDateFin, dateTimeFormatter);
+                    }
+                }
+            }
+        }
+
+        private void log(int index, String title, boolean withDateFin, DateTimeFormatter dateTimeFormatter) {
+            if (withDateFin) {
+                LOGGER.debug("{} {} {} value {}", title, periodes.get(index).dateDebut.format(dateTimeFormatter),
+                        periodes.get(index).dateFin.format(dateTimeFormatter), datas.get(index));
+            } else {
+                LOGGER.debug("{} {} value {}", title, periodes.get(index).dateDebut.format(dateTimeFormatter),
+                        datas.get(index));
+            }
+        }
```

Comment:
Please move this logic into LinkyHandler instead.

Reply:
I will but I don't understand why...

Reply:
Done

Reply:
The reason I suggested moving the logic is because you want to minimize the amount of logic you are doing in DTO classes since the null checker doesn't check dto classes.